packages/sanity/src/structure/comments/plugin/field/CommentsField.tsx

```diff
@@ -29,9 +30,9 @@ const HIGHLIGHT_BLOCK_VARIANTS: Variants = {
 }
 
 export function CommentsField(props: FieldProps) {
-  const isEnabled = useCommentsEnabled()
+  const {enabled} = useCommentsEnabled()
 
-  if (!isEnabled) {
+  if (!enabled) {
```

Comment:
Since we don't render the `CommentFieldInner` component if `enabled` is `false` – we won't show any comment field action on the fields if the enabled value is: `{enabled: false, reason: 'upsell'}`. 

I think it would be nice to create a separate field component that is used for upsell. Something like:

```ts
export function CommentsField(props: FieldProps) {
  const {enabled, reason} = useCommentsEnabled()

  // Comments upsell field component
  if (!enabled && reason === 'upsell') {
    return <CommentsFieldUpsell {...props} />
  }

  // Comments enabled field component
  if (enabled) {
    return <CommentFieldInner {...props} />
  }

  // Default field component without comments
  return props.renderDefault(props)
}
```

That is, the `CommentsFieldInnerUpsell` handles opening the inspector / dialog.

Reply:
It will require to duplicate all the logic already added into the CommentsInput component, as the only thing that changes is the handleClick action when the input has no comments.
If it has comments, it does the same as before (opens the inspector and scrolls into view).

Reply:
however, I could pass the value to the <CommentsFieldInner> by props if we want to avoid using the `useCommentsEnabled` hook in it